internal/service/domain/vehicle/vehicle_test.go

```diff
@@ -0,0 +1,386 @@
+
+		// when
+		result, err := svc.Update(ctx, vehicleID, input)
+
+		// then
+		assert.Nil(t, result)
+		assert.EqualError(t, err, handleError(storage.ErrVehicleNotFound).Error())
+	})
+
+	t.Run("should return an error when the update fails", func(t *testing.T) {
+		vehicleRepo := storageMock.NewMockVehicleRepository(t)
+		svc := NewVehicleService(vehicleRepo)
+
+		expectedErr := errors.New("failed to update cluster")
+		expectedVehicle := getTestVehicles()[0]
+
+		vehicleRepo.EXPECT().GetByID(
+			ctx,
+			vehicleID,
+		).Return(expectedVehicle, nil)
+
+		vehicleRepo.EXPECT().Update(
+			ctx,
+			vehicleID,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+		).Return(nil, expectedErr)
+
+		// when
+		result, err := svc.Update(ctx, vehicleID, input)
+
+		// then
+		assert.Nil(t, result)
+		assert.EqualError(t, err, handleError(expectedErr).Error())
+	})
+}
+
+func TestVehicleService_Delete(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("should successfully delete a vehicle", func(t *testing.T) {
+		vehicleRepo := storageMock.NewMockVehicleRepository(t)
+		svc := NewVehicleService(vehicleRepo)
+
+		id := int32(1)
+
+		vehicleRepo.EXPECT().GetByID(ctx, id).Return(getTestVehicles()[0], nil)
+		vehicleRepo.EXPECT().Delete(ctx, id).Return(nil)
+
+		// when
+		err := svc.Delete(ctx, id)
+
+		// then
+		assert.NoError(t, err)
+	})
+
+	t.Run("should return error if vehicle not found", func(t *testing.T) {
+		vehicleRepo := storageMock.NewMockVehicleRepository(t)
+		svc := NewVehicleService(vehicleRepo)
+
+		id := int32(1)
+		expectedErr := storage.ErrEntityNotFound
+		vehicleRepo.EXPECT().GetByID(ctx, id).Return(nil, expectedErr)
+
+		// when
+		err := svc.Delete(ctx, id)
+
+		// then
+		assert.Error(t, err)
+		assert.EqualError(t, err, handleError(expectedErr).Error())
+	})
+
+	t.Run("should return error if deleting vehicle fails", func(t *testing.T) {
+		vehicleRepo := storageMock.NewMockVehicleRepository(t)
+		svc := NewVehicleService(vehicleRepo)
+
+		id := int32(4)
+		expectedErr := errors.New("failed to delete")
+
+		vehicleRepo.EXPECT().GetByID(ctx, id).Return(getTestVehicles()[0], nil)
+		vehicleRepo.EXPECT().Delete(ctx, id).Return(expectedErr)
+
+		// when
+		err := svc.Delete(ctx, id)
+
+		// then
+		assert.Error(t, err)
+		assert.EqualError(t, err, handleError(expectedErr).Error())
+	})
+}
+
+func TestReady(t *testing.T) {
+	t.Run("should return true if the service is ready", func(t *testing.T) {
+		vehicleRepo := storageMock.NewMockVehicleRepository(t)
+		svc := NewVehicleService(vehicleRepo)
+
+		// when
+		ready := svc.Ready()
+
+		// then
+		assert.True(t, ready)
+	})
+
+	t.Run("should return false if the service is not ready", func(t *testing.T) {
+		svc := NewVehicleService(nil)
+
+		// when
+		ready := svc.Ready()
+
+		// then
+		assert.False(t, ready)
+	})
+}
+
+func getTestVehicles() []*entities.Vehicle {
+	now := time.Now()
+
+	return []*entities.Vehicle{
```

Comment:
Maybe have the "fake" data in a seperate file to have this more organized?

Reply:
I only use this fake data in this file.